manifests/service.pp

```diff
@@ -0,0 +1,30 @@
+# A description of what this class does
+#
+# @summary A short summary of the purpose of this class
+#
+# @example
+#   include puppetwebhook::service
+class puppetwebhook::service {
+  include systemd
+
+  systemd::unit_file { 'puppet_webhook.service':
+    ensure  => 'present',
+    content => epp('puppetwebhook/puppet_webhook.service.epp', {
+        'user'            => $puppetwebhook::app_cfg['user'],
+        'webhook_bin'     => $puppetwebhook::binfile,
+        'webhook_pidfile' => $puppetwebhook::server_cfg['pidfile']
+      }
+    ),
+  }
+
+  service { 'puppet_webhook':
+    ensure     => running,
+    enable     => true,
+    hasstatus  => true,
+    hasrestart => true,
+    require    => [
+      Package['puppet_webhook'],
```

Comment:
Better to define relationships between the install, config and service classes init.pp?

Reply:
(and currently, a config change doesn’t restart the service)

Reply:
I'm using resource relationships for the express purpose that there will be options allowing users to **not** manage the package or service of puppet_webhook if they would prefer to manage it themselves or in some other manner. If we use chaining arrows for resource relationships with said logic, we'd have to write code like this:

``` puppet
if $manage_service {
  Class['puppetwebhook::config']
  ~> Class['puppetwebhook::service']
} elsif $manage_package {
  Class['puppetwebhook::package']
  -> Class['puppetwebhook::config']
} elsif $manage_package && $manage_service {
  Class['puppetwebhook::package']
  -> Class['puppetwebhook::config']
  ~> Class['puppetwebhook::service']
} 
```

Which, to me, is a lot of code duplication for something that can be simplified with local logic in the classes. I think chaining arrows for relationships are great for profiles or company internal modules, but for FOSS modules that are meant to be more like libraries for profiles and internal modules, then chaining arrow based relationships can be limiting (depending on the context)